src/com/owncloud/android/ui/adapter/FileListListAdapter.java

```diff
@@ -344,6 +321,48 @@ public View getView(int position, View convertView, ViewGroup parent) {
         return view;
     }
 
+    private class ThumbailLoaderTask extends AsyncTask<Object, Void, Bitmap> {

```

Comment:
Refactoring the original block in a reusable item is really needed, great point. 

At first I didn't understand the new flow, but in a second thought, seems that makes sense.

Anyway, is not there an easier way to write it? Maybe merging both ```ThumbnailLoaderTask``` and ```ThumbnailGenerationTask``` in a single ```AsyncTask```? Just wondering, didn't think it in depth.